transcom/mymove
# INT B-18916 create customer and okta profile as office user (MERGE AFTER B-18915)

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18916)

## Summary

This PR does contain the code of a currently open PR to use the customer form to ensure functionality. 
[B-18915](https://github.com/transcom/mymove/pull/12248)

A requirement of [E-04903](https://www13.v1host.com/USTRANSCOM38/Epic.mvc/Summary?oidToken=Epic:888544) is for a SC to be able to create a MilMove account for a customer. 

Using the form in the PR above, we can call the backend to do three things:
1. Create the user
2. Create the service member
3. Create the Okta profile (if chosen to do so)

This PR does the following:
- creates a new GHC API endpoint, `POST /createCustomerWithOktaOption`
- adds okta user creation API url to `provider.go`
- adds handler for `createCustomerWithOktaOption` endpoint
- adds function within handler to create the okta user
- adds model to payload conversion in `model_to_payload.go`
- add Okta Go models for profile creation
- minor UI formatting and changes to better handle backend changes
- add flash message to search page and display on successful creation of a customer

### How to test

1. Fire up `client` and `server`
2. Login as a service counselor
3. Search for a user that doesn't exist and click the `Add Customer` button at the bottom of the page
4. You should now see the form to create a customer (click "yes" for okta account creation)
5. Fill it out and call the API (prob best to have your debugger turned on)
6. Make sure all values are being passed into the backend and go through the flow
7. An okta account should now be created and you should get an activation email if you were using the `user+variation@gmail.com` method
8. Log out as a service counselor
9. Log into MilMove as a customer and click the `Local Sign-in` option
10. You should now see the email for the user you just created in the list
11. You can sign in as that user and create a move if ya want
